### PR TITLE
Update QUICHE tarball to 2019-03-08 snapshot of QUICHE master branch.

### PR DESCRIPTION
**Pull request by mpwarres**
Description:
Update QUICHE tarball to static copy of
https://quiche.googlesource.com/quiche/+archive/348de99ee1f13e6747da3695387a2c78610f5f62.tar.gz

Note that with this PR, we are now using the master branch of QUICHE, as
opposed to the previous (temporary) envoy-integration branch.

Apply minor code modifications needed to adapt to changes in QUICHE.

Tested via:
$ bazel test //test/extensions/quic_listeners/quiche/... @com_googlesource_quiche//... --test_output=all --define quiche=enabled

Risk Level: low (code not used yet)
Testing: Unit tests, see description
Docs Changes: none
Release Notes: none
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: source/extensions/quic_listeners/quiche/platform/quic_test_impl.h [hunk 1]
```diff
@@ -18,3 +22,8 @@ class ScopedEnvironmentForThreadsImpl {};
 using QuicTestImpl = ::testing::Test;
 
 template <class T> using QuicTestWithParamImpl = ::testing::TestWithParam<T>;
+
+inline std::string QuicGetTestMemoryCachePathImpl() {
+  PANIC("Not implemented");
```
Comment (wu-bin) [inline source/extensions/quic_listeners/quiche/platform/quic_test_impl.h]: Can we add a TODO for this?

Comment (mpwarres) [reply source/extensions/quic_listeners/quiche/platform/quic_test_impl.h]: Sure, done.

File: source/extensions/quic_listeners/quiche/platform/quic_test_impl.h [hunk 2]
```diff
@@ -18,3 +22,8 @@ class ScopedEnvironmentForThreadsImpl {};
 using QuicTestImpl = ::testing::Test;
 
 template <class T> using QuicTestWithParamImpl = ::testing::TestWithParam<T>;
+
+inline std::string QuicGetTestMemoryCachePathImpl() {
+  PANIC("Not implemented"); // TODO(mpwarres): implement
```
Comment (mattklein123) [inline source/extensions/quic_listeners/quiche/platform/quic_test_impl.h]: nit: NOT_IMPLEMENTED (and I don't think you need the return on the next line)

Comment (mpwarres) [reply source/extensions/quic_listeners/quiche/platform/quic_test_impl.h]: Done, thanks

**Comment by mattklein123** [comment]
Needs format fix.

/wait

**Comment by mpwarres** [comment]
> Needs format fix.
Done.

**Pull request merged**